File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
 Stack<Character> st = new Stack<>();
        
        for(int i=0; i<s.length(); i++){
            char point = s.charAt(i);
            if(point == ')'){
                if(st.empty()){
                    answer = false;
                    break;
                }
                st.pop();
            }
            else{
                st.push(point);
            }
        }
        
        if(!st.empty()){
            answer = false;
        }

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    
    for (const char of s) {
        if (char === '(') {
            stack.push(char);   
        }else {
            if (stack.length && stack[stack.length - 1] === '(') {
                stack.pop();
            }else {
                return false;
            }
        }
    }
    return stack.length ? false : true;
}